src/gcodePlanner.cpp

```diff
@@ -448,6 +448,56 @@ void GCodePlanner::addLinesByOptimizer(Polygons& polygons, GCodePathConfig* conf
     }
 }
 
+void GCodePlanner::spiralizeWallSlice(GCodePathConfig* config, PolygonRef wall, PolygonRef last_wall, const int seam_vertex_idx, const int last_seam_vertex_idx)
+{
+    const Point origin = (last_seam_vertex_idx >= 0) ? last_wall[last_seam_vertex_idx] : wall[seam_vertex_idx];
+    addTravel_simple(origin);
+
+    const int n_points = wall.size();
+    Polygons last_wall_polygons;
+    last_wall_polygons.add(last_wall);
+    const int max_dist2 = config->getLineWidth() * config->getLineWidth() * 4; // (2 * lineWidth)^2;
+
+    double total_length = 0.0; // determine the length of the complete wall
+    Point p0 = origin;
+    for (int index = 1; index <= n_points; ++index)
+    {
+        const Point& p1 = wall[(seam_vertex_idx + index) % n_points];
+        total_length += vSizeMM(p1 - p0);
+        p0 = p1;
+    }
+
+    if (total_length == 0.0)
+    {
+        // nothing to do
+        return;
+    }
+
+    // extrude to the points following the seam vertex
+    // the last point is the seam vertex as the polygon is a loop
+    double wall_length = 0.0;
+    p0 = origin;
+    for (int index = 1; index <= n_points; ++index)
+    {
+        // p is a point from the current wall polygon
+        const Point& p = wall[(seam_vertex_idx + index) % n_points];
+        wall_length += vSizeMM(p - p0);
+        p0 = p;
+
+        // last_p is p shifted onto last_wall
+        Point last_p(p);
+        if (last_wall.inside(last_p, false))
```

Comment:
You should replace these checks by using `PolygonUtils::findClosest(p, last_wall)` and then seeing whether the distance is larger than the max_dist.
I think that's more verbose and more direct for what you are trying to do.